src/nvim/ex_cmds.c

```diff
@@ -5413,7 +5419,10 @@ helptags_one (
    */
   STRCPY(NameBuff, dir);
```

Comment:
You have `dirlen` var and it is already protected by `STRLCPY(NameBuff, dir, MAXPATHL) >= MAXPATHL` above, thus I would simply use `memcpy(NameBuff, dir, (size_t) dirlen + 1)`.